### PR TITLE
allow jil to skip .browser.js test files

### PR DESCRIPTION
**Pull request by metal-messiah**
### Overview
This PR updates JIL to allow `.browser.js` test files to be skipped based on spec rules using `browserMatcher`.  These are not able to be read from the file itself in browser tests, so spec files are now used to export the rules and supply to JIL when preprocessing.

Spec files exist in a 1:1 relationship to tests.  Not all tests need spec files.  These tests with no spec file will run in all spun-up browser instances.  Test files with a spec file will only run in browser instances that match the browserMatcher class exported in the spec.

Functional tests remain untouched.

### CODE REVIEW

File: tools/jil/runner/index.js [hunk 1]
```diff
@@ -77,8 +77,14 @@ function loadDefaultFiles(cb) {
 function loadFiles(testFiles, cb) {
   for (let file of testFiles) {
     file = resolve(process.cwd(), file)
-    if (file.slice(-11) === '.browser.js' || (file.includes('/browser/') && file.endsWith('.test.js'))) {
-      loadBrowser(testDriver, file)
+    if (file.slice(-11) === '.browser.js') {
+      let spec
+      try {
+        spec = require(file.replace('.browser.', '.spec.'))
```
Comment (metal-messiah) [inline tools/jil/runner/index.js:83]: This line in the runner now checks for a spec file to include in the `loadBrowser()` call that stages the browser tests. The spec file must be in the same directory as the test, and share the same name as the test (with spec.js suffix)

File: tests/browser/add-event-listener-options.browser.js [hunk 2]
```diff
@@ -4,10 +4,8 @@
  */
 
 const jil = require('jil')
-let matcher = require('../../tools/jil/util/browser-matcher')
```
Comment (metal-messiah) [inline tests/browser/add-event-listener-options.browser.js:7]: You do not need to remove these lines from the test for it to work, as they are not evaluated anyway.  But these could be removed for clarity in all browser test files if desired.

File: tests/browser/add-event-listener-options.spec.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+let matcher = require('../../tools/jil/util/browser-matcher')
+module.exports = matcher.withFeature('addEventListenerOptions')
```
Comment (metal-messiah) [inline tests/browser/add-event-listener-options.spec.js:2]: Example of what a `*.spec.js` file should export

**Comment by cwli24** [comment]
+Noting that as a side effect, **all** existing `.test.js` files in browser tests folder are changed to `.browser.js`.

**Review by cwli24 — COMMENTED** [review]
👍+Noting that as a side effect, **all** existing `.test.js` files in browser tests folder are changed to `.browser.js`.

**Pull request merged**